src/authenticators/unix_peer_credentials_authenticator/mod.rs

```diff
@@ -143,7 +143,8 @@ mod test {
             .authenticate(&req_auth, conn_metadata)
             .expect("Failed to authenticate");
 
-        assert_eq!(application.identity.name, get_current_uid().to_string());
+        let current_uid: uid_t = unsafe { getuid() };
```

Comment:
This is because we are calling the libc system call getuid. As you can see in https://man7.org/linux/man-pages/man2/geteuid.2.html , this function is always successful and does not modify errno. 

Under the hood, the replaced crate was doing this when calling get_current_uid, as you can see here:
https://docs.rs/users/0.11.0/src/users/base.rs.html#552-557